api/controllers/shabads.js

```diff
@@ -438,64 +437,91 @@ const getShabad = (ShabadIDQ, sinceDate = null) =>
           parameters.push(sinceDate);
         }
 
-        let multipleShabad = '';
+        let multipleShabadOrder = '';
         if (ShabadIDQLength > 1) {
-          multipleShabad = `FIELD(ShabadID, ${tokens}),`;
+          multipleShabadOrder = `FIELD(ShabadID, ${tokens}),`;
           parameters.push(...ShabadIDQ);
         }
 
         const q = `SELECT ${allColumns} ${allFrom}
                     WHERE s.ShabadID IN (${tokens}) ${allColumnsWhere} ${sinceQuery}
-                    ORDER BY ${multipleShabad} v.ID ASC`;
+                    ORDER BY ${multipleShabadOrder} v.ID ASC`;
 
         conn
           .query(q, parameters)
-          .then(rows => {
-            if (rows.length > 0) {
-              const shabadInfo = {
-                shabadId: rows[0].ShabadID,
-                pageNo: rows[0].PageNo,
-                source: lib.getSource(rows[0]),
-                raag: lib.getRaag(rows[0]),
-                writer: lib.getWriter(rows[0]),
+          .then(async rows => {
+            if (rows.length > 0 && ShabadIDQLength === 1) {
+              // single shabad
+              resolve(getShabadSingle(rows));
```

Comment:
This looks kind of weird to me and kinda surprised it works? like resolving a promise with a promise? I guess it should work, but maybe a more readable way would be:

const retShabad = await getShabadSingle(rows);
resolve(retShabad);